businessRules/donations.js

```diff
@@ -1,54 +1,67 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { findIdea } = require('../validations/ideas');
+const {
+    validateSum,
+    findIdeaWithTotalDonationSum,
+} = require('../validations/donations.js');
+const { findUser } = require('../validations/users');
+const { getUsers } = require('../utils/storage');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationByUnregisteredUser(sum, userId, ideaId) {
+    findIdea(ideaId);
+    validateSum(sum);
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
+    let users = getUsers();
+    const user = users.find((user) => userId === user.id);
```

Comment:
Can you use `getUser` function here?

Reply:
Updated this function I don't need those users, only 1 user.  So fixed.